compiler/modules/front.syntax/parseExpr.bal

```diff
@@ -497,7 +497,7 @@ function parseSimpleConstExpr(Tokenizer tok) returns SimpleConstExpr|err:Syntax
     if t == "-" {
         Position opPos = tok.currentStartPos();
         check tok.advance();
-        IntLiteralExpr operand = check parseIntLiteralExpr(tok);
+        NumericLiteralExpr operand = check parseNumericLiteralExpr(tok);
```

Comment:
This was a bug (match pattern with - and floating point literal) There should be a test case.

Reply:
Added match pattern tests cases which passes. Does there need to be any changes made for this?

Reply:
Should we have a condition to use either parseNumericLiteralExpr or parseIntLiteralExpr depending on the use case for now here?

Reply:
Why?

Reply:
Sorry I was confused about the comment earlier, added test cases instead.